src/libsyntax/attr.rs

```diff
@@ -877,10 +886,38 @@ pub fn find_repr_attrs(diagnostic: &Handler, attr: &Attribute) -> Vec<ReprAttr>
                     if let Some(h) = hint {
                         acc.push(h);
                     }
+                } else if let Some((name, value)) = item.name_value_str() {
```

Comment:
Not sure anything else does that already, but for the record, it's a `MetaItemKind::List` (`.meta_item_list()` on `Attribute`) with a single `NestedMetaItemKind::Literal` element (`.literal()` on `NestedMetaItem`), that's a `LitKind::Int(16, LitIntType::Unsuffixed)`.